lightning-invoice/src/payment.rs

```diff
@@ -346,36 +361,74 @@ where
 
 	/// Pays the given [`Invoice`], caching it for later use in case a retry is needed.
 	///
-	/// You should ensure that the `invoice.payment_hash()` is unique and the same payment_hash has
-	/// never been paid before. Because [`InvoicePayer`] is stateless no effort is made to do so
-	/// for you.
+	/// `invoice.payment_hash()` is used as the [`PaymentId`], which ensures idempotency as long as
+	/// the payment is still pending. Once the payment completes or fails, you must ensure that a
+	/// second payment with the same [`PaymentHash`] is never sent.
 	pub fn pay_invoice(&self, invoice: &Invoice) -> Result<PaymentId, PaymentError> {
 		if invoice.amount_milli_satoshis().is_none() {
 			Err(PaymentError::Invoice("amount missing"))
 		} else {
-			self.pay_invoice_using_amount(invoice, None)
+			let payment_id = PaymentId(invoice.payment_hash().into_inner());
```

Comment:
Could use delegate to `pay_invoice_with_id` to avoid repeating the logic? Likewise for other method pairs.